verifact/forum/schema.py

```diff
@@ -118,7 +126,53 @@ def mutate_and_get_payload(
 
         return AnswerCreate(answer=answer)
 
+class VoteUpdate(ClientIDMutation):
+    vote = Field(VoteNode)
+
+    class Input:
+        credible = Boolean() # if null, remove vote!
+        answer_id = ID()
+
+    @login_required
+    def mutate_and_get_payload(
+        self,
+        info,
+        answer_id,
+        **kwargs
+    ):
+        credible = kwargs.get('credible',None)
+        vote=None
+        viewer = info.context.user
+        try: # control flow could have been simpler -> if exists, delete, then create. but chose to do it this way for atomicity.
+            intansid = Node.get_node_from_global_id(
+                info,answer_id,only_type=AnswerNode
+            ).id
+            vote = Vote.objects.get(user=viewer,answer=intansid) # throws ObjectDoesNotExist if none found
```

Comment:
Might be clearer to put only this line inside the `try` block. That way, you can delete this comment without causing any confusion.